### PR TITLE
Add new keyboard: Sparrow62

### PR DESCRIPTION
**Pull request by 74th**
## Description

Adding a new Keyboard: Sparrow62.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/sparrow62/readme.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Sparrow62
+
+Sparrow62 inspired from Lily58.
+
+https://booth.pm/ja/items/2525427
+
+Build Guide: https://github.com/74th/sparrow62-buildguide
```
Comment (zvecr) [inline keyboards/sparrow62/readme.md:7]:
```suggestion
# Sparrow62

Sparrow62 inspired from Lily58.

* Keyboard Maintainer: [Atsushi Morimoto](https://github.com/74th)
* Hardware Availability: https://booth.pm/ja/items/2525427
* Build Guide: https://github.com/74th/sparrow62-buildguide

Make example for this keyboard (after setting up your build environment):

    make sparrow62:default

Flashing example for this keyboard:

    make sparrow62:default:flash

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/sparrow62/keymaps/74th/keymap.c [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                              _______,_______,_______,_______,      _______,_______,_______,KC_DEL
+    //                       \-------+-------+-------+-------/     \-------+-------+-------+-------/
+    )
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    if (record->event.pressed) {
+        switch (keycode) {
+            case LOCK:
+                SEND_STRING(SS_DOWN(X_LGUI) SS_DOWN(X_LCTRL) SS_TAP(X_Q) SS_UP(X_LGUI) SS_UP(X_LCTRL));
```
Comment (fauxpark) [inline keyboards/sparrow62/keymaps/74th/keymap.c:79]:
```suggestion
                tap_code16(C(G(KC_Q)));
```

File: keyboards/sparrow62/keymaps/74th/keymap.c [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                return false;
+            case ESC_MHEN:
+                SEND_STRING(SS_TAP(X_INT5) SS_TAP(X_ESCAPE));
```
Comment (fauxpark) [inline keyboards/sparrow62/keymaps/74th/keymap.c:82]:
```suggestion
                tap_code(KC_INT5);
                tap_code(KC_ESC);
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/sparrow62/keymaps/74th/keymap.c (hunk 3 above)
Comment (74th) [reply keyboards/sparrow62/keymaps/74th/keymap.c:82]: Thanks!

**Review by fauxpark — APPROVED** [review]

**Comment by 74th** [comment]
Thank you for reviews. Sorry, I added 1 commit to fix typo.

File: keyboards/sparrow62/keymaps/74th/keymap.c [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+                              _______,_______,_______,_______,      _______,_______,_______,KC_DEL
+    //                       \-------+-------+-------+-------/     \-------+-------+-------+-------/
+    )
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    if (record->event.pressed) {
+        switch (keycode) {
+            case LOCK:
+                tap_code16(C(G(KC_Q)));
```
Comment (drashna) [inline keyboards/sparrow62/keymaps/74th/keymap.c:79]: you can actually use the `C(G(KC_Q))` in the keymap directly, you don't need a macro for this.

Comment (74th) [reply keyboards/sparrow62/keymaps/74th/keymap.c]: That's right. I fixed it that way.

**Review by drashna — APPROVED** [review]

**Pull request merged**